irctest/cases.py

```diff
@@ -388,10 +397,11 @@ def joinChannel(self, client, channel):
         joined = False
         while not joined:
             for msg in self.getMessages(client):
-                # todo: also respond to cannot join channel numeric
-                if msg.command.upper() == 'JOIN' and 0 < len(msg.params) and msg.params[0].lower() == channel.lower():
+                if msg.command == 'JOIN' and 0 < len(msg.params) and msg.params[0].lower() == channel.lower():
```

Comment:
why did you remove `.upper()`?

Reply:
I actually assumed, incorrectly, that the message parser did this already. But is it necessary? AFAICT `join` is not a valid IRC command.